src/Symfony/Component/PropertyInfo/PropertyInitializableExtractorInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\PropertyInfo;
+
+/**
+ * Guesses if the property can be initialized through the constructor.
+ *
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+interface PropertyInitializableExtractorInterface
+{
+    /**
+     * Is the property initializable?
+     */
+    public function isInitializable(string $class, string $property, array $context = array()): ?bool;
```

Comment:
What's the use-case for returning `null`? Also, shouldn't we explicit within the method name that we are talking about the constructor in here? `isInitializableViaConstructor` or something like that?

Reply:
@teohhanhui is right. It means "this extractor cannot guess", try the next registered in the chain. It's mandatory to keep this behavior for consistency with existing interfaces and to provide an extension point.

Reply:
@sroze I prefer the current name, `isInitializableViaConstructor` is too verbose and brings nothing much.

Reply:
Then we need to rename the interface too?

Reply:
I wouldn't, the PhpDoc that you've put is enough (to me).

Reply:
"Property initializable" indeed makes no sense to me. Perhaps `ConstructorInitializableExtractorInterface::isPropertyInitializable`?

Reply:
`PropertyInitializableViaConstructorExtractorInterface:: isInitializable`?

Reply:
I guess we could just stick with the current names. Lol...

Reply:
@sroze @fabpot ok for you to keep this current names?